gax/src/main/java/com/google/api/gax/rpc/ApiException.java

```diff
@@ -59,4 +68,43 @@ public boolean isRetryable() {
   public StatusCode getStatusCode() {
     return statusCode;
   }
+
+  /**
+   * Returns the reason of the exception. This is a constant value that identifies the proximate
+   * cause of the error.
+   */
+  public String getReason() {
```

Comment:
There are many new methodsadded to this very core `ApiException` all of which seem to be proxying `ErrorDetails` object. Please storngly consider just exposing `ErrorDetails` object from the exception instead of adding these methods.

Reply:
This is one of the requirement from the design, that we want to promote all fields from `ErrorInfo` to the top level, because `ErrorInfo` is the most important/useful one. If we just return `ErrorDetails` which includes all error messages, it might make end users don't know where to start. One comprise could be return `ErrorInfo` directly instead of return three fields separately. 
I'd like to get some input from @summer-ji-eng on this topic as well.

Reply:
I don't think that these kind of decisions are in scope of cross-language design, because the exact way things are supposed to be presented to the user must be language-idiomatic, and thus may differ between the languages. Also, unless @summer-ji-eng disagrees, I would be surprised that the original design intended to dictate such implementation-specific things like this. I.e. please strongly consider removing those methods, as they really contaminate the surface of the class and do not provide much value

Reply:
supposed to be presented to the user must be language-idiomatic, and thus may differ between the languages. Also, unless @summer-ji-eng disagrees, I would be surprised that the original design intended to dictate such implementation-specific things like this. I.e. please strongly consider removing those methods, as they really contaminate the surface of the class and do not provide much value